src/ebay-toggle-button-group/toggle-button-group.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {
+    ReactElement,
+    cloneElement,
+    useState,
+    useCallback,
+    useEffect
+} from 'react'
+import cx from 'classnames'
+import { filterByType } from '../common/component-utils'
+import {
+    ToggleButtonEvent,
+    ToggleButtonProps
+} from '../ebay-toggle-button/types'
+
+import { EbayToggleButton } from '../ebay-toggle-button'
+import { ToggleButtonGroupProps } from './types'
+
+const EbayToggleButtonGroup: React.FC<ToggleButtonGroupProps> = ({
+    a11yText,
+    a11yLabelId,
+    layoutType = 'minimal',
+    variant = 'checkbox',
+    children,
+    columnsMin,
+    columnsXS,
+    columnsSM,
+    columnsMD,
+    columnsXL,
+    onChange,
+    className,
+    ...rest
+}) => {
+    const buttons = filterByType(children, EbayToggleButton)
+    const [pressedButtons, setPressedButtons] = useState({})
+    const handleToggle = useCallback(
+        (toggleEvent, index: number) => {
+            setPressedButtons((prevState) => {
+                let newState = {}
+
+                switch (variant) {
+                    case 'checkbox':
+                        newState = { ...prevState }
+                        newState[index] = !prevState[index]
+                        break
+                    case 'radio-toggle':
+                        newState[index] = !prevState[index]
+                        break
+                    case 'radio':
+                        newState[index] = true
+                        break
+                    default:
+                        break
+                }
+                onChange(toggleEvent, newState)
+                return newState
+            })
+        },
+        [variant, onChange]
+    )
+
+    useEffect(() => {
+        setPressedButtons(
```

Comment:
Lets use the initial state instead of using a use effect. So instead of starting with `{}` we should start with this logic. `useEffect` should be avoided as much as possible and the initial state will help here: